android/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java

```diff
@@ -142,25 +103,17 @@ private void updateSurfaceView() {
                 ((SurfaceView)view).setSecure(true);
             }
         } else {
+            setUseTextureView(true);
             view = new TextureView(context);
             // Support opacity properly:
             ((TextureView) view).setOpaque(false);
         }
         view.setLayoutParams(layoutParams);
-
-        surfaceView = view;
-        if (layout.getChildAt(0) != null) {
-            layout.removeViewAt(0);
-        }
-        layout.addView(surfaceView, 0, layoutParams);
-
-        if (this.player != null) {
-            setVideoView();
-        }
+        addView(view, 0, layoutParams);
```

Comment:
Is there a view leak here ?
you should remove the existing view if it exists ! ?